python/lsst/daf/butler/direct_butler/_direct_butler.py

```diff
@@ -1299,13 +1308,22 @@ def retrieveArtifacts(
         overwrite: bool = False,
     ) -> list[ResourcePath]:
         # Docstring inherited.
-        return self._datastore.retrieveArtifacts(
+        outdir = ResourcePath(destination)
+        paths, id_map, info_map = self._datastore.retrieveArtifacts(
             refs,
-            ResourcePath(destination),
+            outdir,
             transfer=transfer,
             preserve_path=preserve_path,
             overwrite=overwrite,
         )
+        # Write the index file.
+        index = ZipIndex.from_artifact_maps(
+            refs, id_map, info_map, ResourcePath(destination, forceDirectory=True)
+        )
+        index_path = outdir.join("_index.json")
```

Comment:
Should it be `ZipIndex.index_name`?

Reply:
In fact that code shouldn't be there at all now that the index writing is generalized in retrieveArtifacts itself and not part of the zipping. Removed.